js/modules/loki_snode_api.js

```diff
@@ -756,73 +756,125 @@ class LokiSnodeAPI {
     }
   }
 
-  async getLnsMapping(lnsName) {
+  async getLnsMapping(lnsName, timeout) {
+    // Returns { pubkey, error }
+    // pubkey is:
+    //      null      when there is confirmed to be no LNS mapping
+    //      undefined when unconfirmed
+    //      string    when found
+    // timeout parameter optional (ms)
+
+    // How many nodes to fetch data from?
+    const numRequests = 5;
+
+    // How many nodes must have the same response value?
+    const numRequiredConfirms = 3;
+
+    let ciphertextHex;
+    let pubkey;
+    let error;
+
     const _ = window.Lodash;
 
     const input = Buffer.from(lnsName);
-
     const output = await window.blake2b(input);
-
     const nameHash = dcodeIO.ByteBuffer.wrap(output).toString('base64');
 
+    // Return value of null represents a timeout
+    const timeoutResponse = { timedOut: true };
+    const timeoutPromise = (cb, interval) => () =>
+      new Promise(resolve => setTimeout(() => cb(resolve), interval));
+    const onTimeout = timeoutPromise(
+      resolve => resolve(timeoutResponse),
+      timeout || Number.MAX_SAFE_INTEGER
+    );
+
     // Get nodes capable of doing LNS
-    const lnsNodes = this.getNodesMinVersion('2.0.3');
-    // randomPool should already be shuffled
-    // lnsNodes = _.shuffle(lnsNodes);
+    let lnsNodes = await this.getNodesMinVersion(
+      window.CONSTANTS.LNS_CAPABLE_NODES_VERSION
+    );
+    lnsNodes = _.shuffle(lnsNodes);
```

Comment:
Looks like you don't need to shuffle, you can just delete this like (was explained by the comment you removed).